internal/services/appservice/function_app_flex_consumption_resource.go

```diff
@@ -847,47 +1204,134 @@ func (r FunctionAppFlexConsumptionResource) Update() sdk.ResourceFunc {
+						}
+						storageAuth.UserAssignedIdentityResourceId = &StorageUserAssignedIdentityID
+					}
+				}
+				model.Properties.FunctionAppConfig.Deployment.Storage.Authentication = &storageAuth
+			}
+
+			if metadata.ResourceData.HasChange("deployment_storage_authentication_type") {
+				storageAuthType := webapps.AuthenticationType(state.DeploymentStorageAuthType)
+				storageAuth := webapps.FunctionsDeploymentStorageAuthentication{
+					Type: &storageAuthType,
+				}
+				if state.DeploymentStorageAuthType == string(webapps.AuthenticationTypeStorageAccountConnectionString) {
+					storageConnStringForFCApp = "DEPLOYMENT_STORAGE_CONNECTION_STRING"
+					if deploymentSaKey == "" {
 						return fmt.Errorf("the storage account access key must be specified when using the storage key based access")
 					}
 					storageAuth.StorageAccountConnectionStringName = pointer.To(storageConnStringForFCApp)
 				} else {
 					storageConnStringForFCApp = ""
-					if state.StorageAuthType == string(webapps.AuthenticationTypeUserAssignedIdentity) {
-						if state.StorageUserAssignedIdentityID == "" {
+					storageConnStringForFcAppValue = ""
+					if state.DeploymentStorageAuthType == string(webapps.AuthenticationTypeUserAssignedIdentity) {
+						if state.DeploymentStorageUserAssignedIdentityID == "" {
 							return fmt.Errorf("the user assigned identity id must be specified when using the user assigned identity to access the storage account")
 						}
-						storageAuth.UserAssignedIdentityResourceId = &state.StorageUserAssignedIdentityID
+						storageAuth.UserAssignedIdentityResourceId = &state.DeploymentStorageUserAssignedIdentityID
 					}
 				}
 				model.Properties.FunctionAppConfig.Deployment.Storage.Authentication = &storageAuth
 			}
 
-			if metadata.ResourceData.HasChange("storage_user_assigned_identity_id") {
+			if !features.FivePointOh() && metadata.ResourceData.HasChange("storage_user_assigned_identity_id") {
 				model.Properties.FunctionAppConfig.Deployment.Storage.Authentication.UserAssignedIdentityResourceId = &state.StorageUserAssignedIdentityID
 			}
 
+			if metadata.ResourceData.HasChange("deployment_storage_user_assigned_identity_id") {
+				model.Properties.FunctionAppConfig.Deployment.Storage.Authentication.UserAssignedIdentityResourceId = &state.DeploymentStorageUserAssignedIdentityID
+			}
+
 			// Note: We process this regardless to give us a "clean" view of service-side app_settings, so we can reconcile the user-defined entries later
-			siteConfig, err := helpers.ExpandSiteConfigFunctionFlexConsumptionApp(state.SiteConfig, model.Properties.SiteConfig, metadata, false, storageString, storageConnStringForFCApp)
+			siteConfig, err := helpers.ExpandSiteConfigFunctionFlexConsumptionApp(state.SiteConfig, model.Properties.SiteConfig, metadata, state.StorageAccountUsesMSI, backendStorageString, storageConnStringForFCApp, storageConnStringForFcAppValue)
```

Comment:
I thought it was a nice move when previously you put each argument on its own line, maybe here too?